giskard/scanner/llm/llm_prompt_injection_detector.py

```diff
@@ -30,89 +35,58 @@ class LLMPromptInjectionDetector(Detector):
-        column_types = dataset.column_types
-
-        prompts = get_all_prompts()
+        generator = PromptInjectionDataLoader(num_samples=self.num_samples)
+        dataset = generator.load_dataset(dataset.column_types)
+        meta_df = generator.all_meta_df
 
+        evaluator = StringMatcher()
         issues = []
-
-        results = self.evaluate_and_group(model, dataset, prompts, features, column_types)
-
-        for group in results.keys():
-            failed_examples = {}
-            cols = []
-            prompts_names = [prompt.name for prompt in results[group]["input_prompt"]]
-            prompts_contents = [prompt.content for prompt in results[group]["input_prompt"]]
-            if all(prompts_names):
-                failed_examples.update({"Prompt Name": prompts_names})
-                cols = ["Prompt Name"]
-
-            failed_examples.update({"Input Prompt": prompts_contents, "Prediction": results[group]["prediction"]})
-            cols += ["Input Prompt", "Prediction"]
-
-            failed_examples_df = pd.DataFrame(failed_examples)[cols]
-            failed_examples_df = failed_examples_df.filter(
-                items=[i for i, v in enumerate(results[group]["failed"]) if v != 0], axis=0
-            )
-
-            failed = sum(results[group]["failed"])
-            if failed == 0:
+        for group in set(generator.groups_mapping):
+            group_idx = meta_df.index[meta_df["group_mapping"] == group].tolist()
+            group_dataset = dataset.slice(group_slice(group_idx=group_idx))
+            group_meta_df = meta_df.iloc[group_idx]
+            evaluation_results = evaluator.evaluate(model, group_dataset, group_meta_df.to_dict("records"))
+            number_of_failed_prompts = len(evaluation_results.failure_examples)
+            if number_of_failed_prompts == 0:
                 continue
-            total = len(results[group]["failed"])
-            metric = failed / total
+            metric = 1 - evaluation_results.passed_ratio
 
             level = IssueLevel.MINOR
             if 0.1 <= metric < self.threshold:
                 level = IssueLevel.MEDIUM
             elif metric >= self.threshold:
                 level = IssueLevel.MAJOR
 
+            group_description = meta_df[meta_df.group_mapping == group].description.to_list()
+            group_deviation_description = meta_df[meta_df.group_mapping == group].deviation_description.to_list()
+            if len(set(group_description)) != 1:
+                raise ValueError(f"{self.__class__.__name__}: There must be only one group description per group.")
+            if len(set(group_deviation_description)) != 1:
+                raise ValueError(
+                    f"{self.__class__.__name__}: There must be only one group description deviation per group."
+                )
+
+            group_description, group_deviation_description = group_description[0], group_deviation_description[0]
```

Comment:
I would skip this check. Also I don’t feel this logic should be here, but if needed inside the loader.